### PR TITLE
update release configuration - use flint

### PR DESCRIPTION
**Pull request by shemogumbe**
Closed https://github.com/microsoftgraph/msgraph-beta-sdk-python/issues/412

### CODE REVIEW

**Review by andrueastman — COMMENTED** [review]
Similar comment to https://github.com/microsoftgraph/msgraph-sdk-python/pull/738

**Review by andrueastman — DISMISSED** [review]

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -1,29 +1,32 @@
...
   pull_request:
     branches: [ main, dev ]
-
-permissions:
-  contents: read
+  workflow_call:
 
 jobs:
-  validate:
-    name: Validate code accuracy
```
Comment (andrueastman) [inline .github/workflows/build.yml:16]: If you remove the name property here, the required check is no longer able to find the status to display.

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -9,7 +9,38 @@ permissions:
...
+    name: Update version
+    runs-on: ubuntu-latest
+    if: startsWith(github.ref, 'refs/tags/v')
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.9
```
Comment (baywet) [inline .github/workflows/publish.yml:22]:
```suggestion
          python-version: 3.12
```

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -9,7 +9,38 @@ permissions:
...
+        run: |
+          git config --local user.email "${{ secrets.GIT_USER_EMAIL }}"
+          git config --local user.name "${{ secrets.GIT_USER_NAME }}"
+          git commit -am "Update version"
+          
+      - name: Push changes
+        uses: ad-m/github-push-action@master
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          branch: ${{ github.ref }}
```
Comment (baywet) [inline .github/workflows/publish.yml:40]:
```suggestion
          git commit -am "chore: Update version"
          git push
```

this way we're ready for release please
and we reduce our dependencies on third party

**Review by baywet — CHANGES_REQUESTED** [review]

File: .github/workflows/publish.yml [hunk 4]
```diff
@@ -24,18 +50,16 @@ jobs:
       - name: Set up Python
         uses: actions/setup-python@v5
         with:
-          python-version: 3.12
-      - name: Install dependencies
+          python-version: 3.9
```
Comment (baywet) [inline .github/workflows/publish.yml:53]:
```suggestion
          python-version: 3.12
```

**Review by baywet — APPROVED** [review]

File: .github/workflows/publish.yml [hunk 5]
```diff
@@ -9,7 +9,38 @@ permissions:
...
+        run: |
+          VERSION=${GITHUB_REF#refs/tags/v}
+          poetry version $VERSION
+      - name: Commit changes
+        run: |
+          git config --local user.email "${{ secrets.GIT_USER_EMAIL }}"
+          git config --local user.name "${{ secrets.GIT_USER_NAME }}"
+          git commit -am "chore: Update version"
+
+      - name: Push changes
```
Comment (baywet) [inline .github/workflows/publish.yml:36]: @shemogumbe why have you reverted the suggested change? can you provide context when you do things like this please?

Comment (shemogumbe) [reply .github/workflows/publish.yml:36]: The idea here is to make a commit with the version change, then push that commit.
The change  was up to the stage of the commit, changes to allow a git push are new.
I think the commit is factored in the latest changes, right?

Comment (baywet) [reply .github/workflows/publish.yml:36]: but why aren't you simply using a git push command in the previous script block?

Comment (baywet) [reply .github/workflows/publish.yml:36]: so am I understanding things correctly that you'll revert this change to: remove the push action step (block below) and add a git push command to the block above?

Comment (shemogumbe) [reply .github/workflows/publish.yml:36]: Sure, on it

**Review by baywet — APPROVED** [review]

**Pull request merged**